### PR TITLE
Remove boost-program-options and fix macOS support.

### PR DESCRIPTION
**Pull request by Yashwants19**
Hi @rcurtin, I have tried to remove `boost-program-options` and also tried to fix `macOS` installation from `MAKEFILE` and also add it to travis-ci.

I have also apply the new auto-generated go-binding from `mlpack-3.4.0` as instructed in `rel/deployment.md`, if you find this incorrect I can revert this last commit.
Let me know what do you think?

Thank you.

### CODE REVIEW

File: linear_regression.go [hunk 1]
```diff
@@ -8,11 +8,7 @@ package mlpack
 */
```
Comment (rcurtin) [inline linear_regression.go:8]: I can't comment on the binary file, but this is the next file. :)  I think that `libmlpack.so.3.3` shouldn't be included. :+1:

File: test_go_binding.go [hunk 2]
```diff
@@ -8,11 +8,7 @@ package mlpack
 */
```
Comment (rcurtin) [inline test_go_binding.go:8]: It might be worth removing this binding too; it may not be useful for users. :+1:

**Review by rcurtin — COMMENTED** [review]
Looks good to me, although I think the .so file was mistakenly included.

Were you able to follow the steps in https://github.com/mlpack/mlpack-go/blob/master/rel/deployment.md exactly?  If you had to do anything different (other than updating the travis config), we should change that too in `deployment.md` for the future when we can manage to automate this. :)

File: linear_regression.go (hunk 1 above)
Comment (Yashwants19) [reply linear_regression.go:8]: Removed this file.

File: test_go_binding.go (hunk 2 above)
Comment (Yashwants19) [reply test_go_binding.go:8]: No problem, we can remove this, I just thought that it was useful for testing the `MAKEFILE` file.

**Comment by Yashwants19** [comment]
> Were you able to follow the steps in https://github.com/mlpack/mlpack-go/blob/master/rel/deployment.md exactly? If you had to do anything different (other than updating the travis config), we should change that too in deployment.md for the future when we can manage to automate this. :)

Yeah I follow the same steps given in `deployment.md`, except I have to remove some of the files `cli_util.go/h`, `range_search.go/h`, etc, which step was not mentioned in `deployment.md`, Should I add this line too in `deployment.md`.

File: test_go_binding.go (hunk 2 above)
Comment (Yashwants19) [reply test_go_binding.go:8]: I think so, after the confirmation on this we can release go-bindings with a release-tag same as mlpack.

Comment (rcurtin) [reply test_go_binding.go:8]: Sorry for the slow response---it's a holiday weekend in the US so I have been away from a computer. :)  Personally I don't have too much of an opinion, so if you think it's useful for users to have for the sake of testing, we can leave it in.

**Comment by rcurtin** [comment]
> > Were you able to follow the steps in https://github.com/mlpack/mlpack-go/blob/master/rel/deployment.md exactly? If you had to do anything different (other than updating the travis config), we should change that too in deployment.md for the future when we can manage to automate this. :)
> 
> Yeah I follow the same steps given in `deployment.md`, except I have to remove some of the files `cli_util.go/h`, `range_search.go/h`, etc, which step was not mentioned in `deployment.md`, Should I add this line too in `deployment.md`.

It looks like you updated `deployment.md` to handle these steps, so that looks good to me. :+1:

Let me know what you think about the test binding, and then I'll go ahead and merge this.

(Also, I'm about to release 3.4.1, so I guess we will do this again :))

File: test_go_binding.go (hunk 2 above)
Comment (Yashwants19) [reply test_go_binding.go:8]: This sounds awesome to have holidays, I guess have you enjoyed the holiday :) :tada:

**Comment by Yashwants19** [comment]
I think so this is ready to merge.

(If you want I am ready to open a PR for 3.4.1 :))

**Comment by Yashwants19** [comment]
Its seems that [#2460](https://github.com/mlpack/mlpack/pull/2460) helps us to reduce the time for building `go-shlib` from 20 min to 11 min. :)

**Comment by rcurtin** [comment]
Sure, if you want to open the 3.4.1 PR too, we can make sure Travis passes and then merge it. :)

**Comment by Yashwants19** [comment]
Hi @rcurtin  I think so, this time `LICENSE.txt` works fine, as this website looks great:
https://pkg.go.dev/mlpack.org/v1/mlpack@v0.0.0-20200907173821-d2b489013df6?tab=overview

**Comment by rcurtin** [comment]
Awesome!  How can we change the listed version of the package to 3.4.0?  If that's something that we can do in this repository, maybe you can handle that in the 3.4.1 update?

**Comment by Yashwants19** [comment]
As I mention previously in [comment](https://github.com/mlpack/mlpack-go/pull/8#discussion_r483370572), release with a tag `v3.4.0` might help us here.

**Comment by rcurtin** [comment]
Oh, sorry, I missed that!  Let me do that.

**Comment by Yashwants19** [comment]
Hi I am really sorry the tag was `v3.4.0` and not `3.4.0`.

**Comment by rcurtin** [comment]
Ahh, easy fix, hang on. :)

**Pull request merged**